rpc/legacyrpc/methods.go

```diff
@@ -3246,7 +3251,8 @@ func sweepAccount(s *Server, icmd interface{}) (interface{}, error) {
 		return nil, err
 	}
 
-	changeSource := makeChangeSourceFromAddress(cmd.DestinationAddress)
+	changeSource := new(changeSourceFromAddress)
+	changeSource.address = cmd.DestinationAddress
```

Comment:
can use `&scriptChangeSource{script, version}` to initialize this after decoding the address.